lib/team_stats.rb

```diff
@@ -12,81 +12,94 @@ def team_info(team_id)
+      wins = games_array.find_all {|game| winning_game_ids(team_id).include?(game.game_id)}.count
       team_games_grouped_by_season[season] = wins.to_f / games_array.count
     end
     team_games_grouped_by_season.max_by {|_, win_percentage| win_percentage}.first
   end
 
   def worst_season(team_id)
-    team_wins = @game_teams.find_all {|game| game.team_id == team_id && game.result == "WIN"}
-    winning_game_ids = team_wins.map {|game| game.game_id}
-    team_games = @games.find_all {|game| game.away_team_id == team_id || game.home_team_id == team_id}
-    team_games_grouped_by_season = team_games.group_by {|game| game.season}
+    team_games_grouped_by_season = find_home_or_away(team_id).group_by {|game| game.season}
     team_games_grouped_by_season.each do |season, games_array|
-      wins = games_array.find_all {|game| winning_game_ids.include?(game.game_id)}.count
+      wins = games_array.find_all {|game| winning_game_ids(team_id).include?(game.game_id)}.count
       team_games_grouped_by_season[season] = wins.to_f / games_array.count
     end
     team_games_grouped_by_season.min_by {|_, win_percentage| win_percentage}.first
   end
 
   def average_win_percentage(team_id)
-    team_wins = @game_teams.find_all {|game| game.team_id == team_id && game.result == "WIN"}.count
     team_losses = @game_teams.find_all {|game| game.team_id == team_id && game.result == "LOSS"}.count
     team_ties = @game_teams.find_all {|game| game.team_id == team_id && game.result == "TIE"}.count
-    total_games = team_wins + team_losses + team_ties
-    (team_wins.to_f / total_games).round(2)
+    total_games = team_wins(team_id).count + team_losses + team_ties
+    (team_wins(team_id).count.to_f / total_games).round(2)
+  end
+
+  def find_all_games_by(team_id)
+    @game_teams.find_all {|game| game.team_id == team_id}
   end
 
   def most_goals_scored(team_id)
-    team = @game_teams.find_all {|game| game.team_id == team_id}
-    x = team.max_by do |game|
+    goals_scored = find_all_games_by(team_id).max_by do |game|
       game.goals
     end
-    x.goals
+    goals_scored.goals
   end
 
   def fewest_goals_scored(team_id)
-    team = @game_teams.find_all {|game| game.team_id == team_id}
-    x = team.min_by do |game|
+    goals_scored = find_all_games_by(team_id).min_by do |game|
       game.goals
     end
-    x.goals
+    goals_scored.goals
+  end
+
+  def find_all_games_by_given_team_id(team_id)
+    @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
   end
 
   def favorite_opponent(team_id)
-    find_by_given_team_id = @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
-    x = find_by_given_team_id.map {|game| game.game_id}
-    y = @game_teams.find_all {|game| x.include?(game.game_id) && game.team_id != team_id}
-    sorted_by_team_id = y.group_by {|game| game.team_id}
+    find_game_ids = find_all_games_by_given_team_id(team_id).map {|game| game.game_id}
```

Comment:
Trying to break this into a helper method makes the spec run very slow. around 50 seconds. so thats why i didn't refactor beyond just the variable name

Reply:
Hey I think that is great. You tried it and it didn't improve the performance. Nice job breaking all of these out.